### PR TITLE
`sage.combinat.words`: Update `# needs`

### PR DESCRIPTION
**Pull request by mkoeppe**
- Cherry-picked from #35095 

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/combinat/words/finite_word.py [hunk 1]
```diff
@@ -6823,18 +6824,18 @@ def colored_vector(self, x=0, y=0, width='default', height=1, cmap='hsv', thickn
 
         EXAMPLES::
 
-            sage: Word(range(20)).colored_vector()                                      # optional - sage.plot
+            sage: Word(range(20)).colored_vector()                                      # needs sage.plot
```
Comment (dcoudert) [inline src/sage/combinat/words/finite_word.py]: block scope

File: src/sage/combinat/words/morphism.py [hunk 2]
```diff
@@ -1099,21 +1101,21 @@ def _matrix_(self, R=None):
 
             sage: fibo = WordMorphism('a->ab,b->a')
             sage: tm = WordMorphism('a->ab,b->ba')
-            sage: Mfibo = matrix(fibo); Mfibo     # indirect doctest                    # optional - sage.modules
+            sage: Mfibo = matrix(fibo); Mfibo     # indirect doctest                    # needs sage.modules
```
Comment (dcoudert) [inline src/sage/combinat/words/morphism.py]: block scope

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**